cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -708,3 +720,21 @@ func (c *machineController) allowedByAutoDiscoverySpecs(r *unstructured.Unstruct
 
 	return false
 }
+
+// Get an infrastructure machine template given its GVR, name, and namespace.
+func (c *machineController) getInfrastructureResource(resource schema.GroupVersionResource, name string, namespace string) (*unstructured.Unstructured, error) {
+	infra, err := c.managementClient.
```

Comment:
The logs you mentioned are because of this client is not using a cache. We should be using a native cache e.g https://github.com/kubernetes-sigs/controller-runtime/blob/88234a8fc9ad7f559a593e098f224dcecf4c6f6f/pkg/cache/internal/informers_map.go#L311

controller runtime gives you the abstractions for it https://github.com/kubernetes-sigs/controller-runtime/issues/615

Reply:
ok, perhaps i'm just not understanding the client well enough. i had thought that by using the dynamic client and adding these types as we discover them that they will be added to an internal cache.

also, we are using client-go afaik and not controller-runtime in here. i would be happy to fix this up so that we are not overloading the kas, i just need some pointers about how to get there using client-go.

Reply:
ok, think i've got this fixed

> It's not about having noisy logs or not,

fwiw, i agree, logs were just a symptom